src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs

```diff
@@ -593,16 +593,19 @@ private void UnregisterFileSystemWatcher()
 
                 _logger.Log(LogLevel.Verbose, "File system changed. File: {0} - Change: {1}", e.Name, e.ChangeType);
 
+                var changedDirectory = Path.GetFullPath(Path.GetDirectoryName(e.FullPath) ?? throw new InvalidOperationException());
```

Comment:
We should just return in the null directory name case. (I'm not sure how this could have in practice though).